File: frontend/src/objects/order/apiTypes/CheckoutResponse.ts
import type { Voucher } from '@/objects/shared/Voucher'

import type { Order } from '../Order'
import type { OrderPriceBreakdown } from '../OrderPriceBreakdown'

export interface CheckoutResponse {
  orders: Order[]
  walletBalance: number
  originalAmount: number
  discountAmount: number
  payableAmount: number
  usedVoucher?: Voucher | null
  priceBreakdown?: OrderPriceBreakdown
}

File: backend/src/order/objects/apiTypes/CheckoutResponse.scala
package delivery.order.objects.apiTypes

import delivery.order.objects.Order
import delivery.promotion.objects.Voucher

final case class CheckoutResponse(
    orders: List[Order],
    walletBalance: Double,
    originalAmount: Double,
    discountAmount: Double,
    payableAmount: Double,
    usedVoucher: Option[Voucher],
    priceBreakdown: delivery.order.objects.OrderPriceBreakdown
)
